miniapps/shifted/dist_solver.hpp

```diff
@@ -70,19 +70,38 @@ class HeatDistanceSolver : public DistanceSolver
    bool transform, vis_glvis;
 };
 
+// A. Belyaev et al: "On Variational and PDE-based Distance Function
+// Approximations", Section 6, DOI:10.1111/cgf.12611.
+class NormalizationDistanceSolver : public DistanceSolver
+{
+private:
+
+   class NormalizationCoeff : public Coefficient
+   {
+   private:
+      ParGridFunction &u;
+
+   public:
+      NormalizationCoeff(ParGridFunction &u_gf) : u(u_gf) { }
+      virtual double Eval(ElementTransformation &T, const IntegrationPoint &ip);
+   };
+
+public:
+   NormalizationDistanceSolver() { }
+
+   void ComputeScalarDistance(Coefficient& u_coeff, ParGridFunction& dist);
```

Comment:
Might be worth commenting here and in the introduction of the miniapp that this solver is supposed to be a computationally cheaper alternative to pLaplacian and is accurate for distance approximations near the zero level set.